File: 3195-separate-black-and-white-balls/separate-black-and-white-balls.cpp
class Solution {
public:
    long long minimumSteps(string s) {
        
        int i=0  , j=s.size()-1;
        long long ans=0;

        while(i<j){
            while(s[i]=='0' and j>i) i++;
            
            while(s[j]=='1' and j>i) j--;

            swap(s[i],s[j]);
            ans+=(j-i);
        }
        return ans;
    }
};

File: 3195-separate-black-and-white-balls/separate-black-and-white-balls.js
/**
 * @param {string} s
 * @return {number}
 */
var minimumSteps = function(s) {
    
    let ans=0;
    let black=0;

    for (let i = 0; i < s.length; i++) {
        if (s[i] === '0') ans += black;
        else black++;
    }

    return ans;
};